### PR TITLE
OKD: Switch to using Centos Stream base container images

### PR DESCRIPTION
**Pull request by sdodson**
Because OpenShift depends on many packages not part of UBI redistributable
content we cannot use RHEL UBI as the base image. Thankfully Centos Stream
should be fully compatible, so lets use that instead.

/cc @LorbusChris @vrutkovs

### CODE REVIEW

**Comment by vrutkovs** [comment]
>Implementing and maintaining support for OKD in individual components shall not require new levels of commitment by engineers

Is this still holds true?

**Review by vrutkovs — CHANGES_REQUESTED** [review]
This needs describe new responsibilities on community / engineers / whoever wants to pick up the task of making new base images, adding PRs to add them to the repo and keeping release repo updated along with triaging if the bug comes from CentOS Stream or also applies to OCP

**Comment by sdodson** [comment]
I think the commitment becomes the need to go ensure that all packages necessary are available in Centos Stream. Beyond that I don't envision there being significant burden on engineers.

**Comment by sdodson** [comment]
It seems like consensus is that there will definitely be a huge cost to switch. The ongoing steady state build pipeline seems like we can manage. The cost will be for anyone who wants to add a new package to OpenShift that's not readily available in Centos Stream or substantially different from that which is available as part of RHEL. OVS coming from RHEL Fast Data Path being substantially ahead of whatever version of openvswitch is available in either Centos Stream or CBS would be a potential example, though it does seem like openvswitch3.2 is readily available in CBS today.

**Comment by sdodson** [comment]
@vrutkovs what else do we need to do before we approve this?

**Comment by vrutkovs** [comment]
I'd recommend running this enhancement on nearest arch meeting, adding approvers meanwhile:
/cc @deads2k @dhellmann @bparees

File: enhancements/okd-on-fedora-coreos-in-prow.md
Comment (dhellmann) [inline enhancements/okd-on-fedora-coreos-in-prow.md:81]: Can we be more explicit about saying that none of the images include any RHEL content?

```suggestion
- [ ] Container images for all components in the release payload are built atop Centos Stream base images most closely matching the majority of OCP images (as of Feb 2024, CS9) and do not contain any content from RHEL builds of any packages.
```

File: enhancements/okd-on-fedora-coreos-in-prow.md
Comment (vrutkovs) [inline enhancements/okd-on-fedora-coreos-in-prow.md:81]: We should add ideas how to enforce it - periodics which can list RPMs and image metadata to prevent the leaks happening again?

>RHEL builds of any packages

I agree on the idea, but I'd prefer the base idea to still be UBI so that the diff would be minimal and we'd have less OKD-specific issues caused by possible "current RHEL" <-> CentOS Stream disrepancies

Comment (sdodson) [reply enhancements/okd-on-fedora-coreos-in-prow.md:81]: Your motivations for using UBI (stated offline), to maintain bug-for-bug compatibility with OCP, cannot be met without including RHEL content which cannot be redistributed. While we could use UBI as the base but Centos Stream for everything else I don't understand the value that provides given it'll never be bug-for-bug compatible. I would like to move forward with this PR since UBI is not valuable.

I agree that we should seek to reduce duplicate efforts of maintenance and I have confidence that work is in progress. I'd like to move forward with this PR as is today.

Comment (vrutkovs) [reply enhancements/okd-on-fedora-coreos-in-prow.md:81]: >While we could use UBI as the base but Centos Stream for everything else I don't understand the value that provides given it'll never be bug-for-bug compatible

I don't mind CentOS Stream used as base. Perhaps using UBI instead would make devs more comfortable with it, but I suppose we can reach an agreement on this.

However its vital to use the same OCP builder if possible/legal - this would help with developer trust (binaries being built with the same compiler as OCP CI) and avoid a costly double rebuild for OKD binaries

Comment (vrutkovs) [reply enhancements/okd-on-fedora-coreos-in-prow.md:81]: This still needs to list who's responsible for generating C9S-based images - I'd prefer for this to happen transparently on ART side if possible

Comment (ausil) [reply enhancements/okd-on-fedora-coreos-in-prow.md]: We cannot use the same builder image as we want the community to write and submit patches, which require them to be able to build. The OCP Builder images have the same restrictions and we can not make them available.

**Comment by ausil** [comment]
Until we have removed RHEL from OKD we can not ship any more OKD releases. It is in our and the community's interests to quickly resolve the issues so we can get OKD working again. It was communicated to the community on February 6th in the Community call https://www.youtube.com/watch?v=P3A1oe3hf0Y

File: enhancements/okd-on-fedora-coreos-in-prow.md
Comment (sdodson) [reply enhancements/okd-on-fedora-coreos-in-prow.md]: @vrutkovs please take a look at the new changes I pushed

Comment (vrutkovs) [reply enhancements/okd-on-fedora-coreos-in-prow.md]: >We cannot use the same builder image as we want the community to write and submit patches

Community can use a CentOS9 or UBI builder, it doesn't have to be bit-to-bit identical to the builder we use in Prow. This would allow us to use less resources when OKD is being built

File: enhancements/okd-on-fedora-coreos-in-prow.md
Comment (vrutkovs) [inline enhancements/okd-on-fedora-coreos-in-prow.md:81]: >OCP Engineering teams will be responsible for addressing bugs reported against OKD

Who's responsible for triaging OKD bugs - both in Jira and in github issues?

Comment (vrutkovs) [reply enhancements/okd-on-fedora-coreos-in-prow.md:81]: >do not contain any content from RHEL builds of any packages

How are we planning to enforce that? There is nothing preventing devs from using internal RPM repos

Comment (sdodson) [reply enhancements/okd-on-fedora-coreos-in-prow.md:81]: I think Github would need to be triaged first by OKD team/volunteers and if necessary OCPBUGS filed. They're just OCP bugs then so they'd be triaged by all the normal teams.

As to enforcement of no RHEL content that will depend on the implementation path chosen.

**Review by dhellmann — COMMENTED** [review]
/approve

**Pull request merged**
